ahnlich/ai/src/cli/server.rs

```diff
@@ -301,3 +309,53 @@ impl SupportedModelArgs {
         writeln!(&mut stdout, "{text}").expect("Failed to write output");
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use regex::Regex;
+    use std::process;
+    use test_case::test_case;
+
+    #[test_case("--db-host", Some("127.0.0.1"), Some("<DB_HOST>"); "with_db_host_option_passed")]
+    #[test_case("--db-port", Some("1369"), Some("<DB_PORT>"); "with_db_port_option_passed")]
+    #[test_case("--db-https", None, None; "with_db_https_option_passed")]
```

Comment:
Nice call on the test but it would then seem even "db-https" should be unusable with --without-db then? i.e no point setting that if you've said without db

Would also ellminate all the options and just be `&str`, `&str`, `&str`

Reply:
Ah okay I see what you mean... still conficts but only with env var names and placeholders